core/src/main/java/nl/nn/adapterframework/jms/JmsSender.java

```diff
@@ -245,22 +239,28 @@ public Message sendMessage(Message message, PipeLineSession session, String soap
 					if (rawReplyMsg==null) {
 						throw new TimeoutException("did not receive reply on [" + replyQueue + "] requestMsgId ["+msg.getJMSMessageID()+"] replyCorrelationId ["+replyCorrelationId+"] within ["+getReplyTimeout()+"] ms");
 					}
-					if(getResponseHeadersList().size() > 0) {
+					StringBuilder receivedJMSProperties = new StringBuilder();
+					if(!getResponseHeadersList().isEmpty()) {
 						Enumeration<?> propertyNames = rawReplyMsg.getPropertyNames();
 						while(propertyNames.hasMoreElements()) {
 							String jmsProperty = (String) propertyNames.nextElement();
 							if(getResponseHeadersList().contains(jmsProperty)) {
 								session.put(jmsProperty, rawReplyMsg.getObjectProperty(jmsProperty));
+								if (log.isDebugEnabled()) {
+									receivedJMSProperties.append(jmsProperty).append(": ").append(rawReplyMsg.getObjectProperty(jmsProperty)).append("; ");
+								}
 							}
 						}
 					}
-					return extractMessage(rawReplyMsg, session, isSoap(), getReplySoapHeaderSessionKey(),soapWrapper);
+					logMessageDetails(rawReplyMsg);
+					log.debug("Received properties: {}", receivedJMSProperties);
```

Comment:
@alisihab here you can see the debug logging for the JMS Properties. It is build up at line 250. 
Are you able to test a dev-build to see if it is sufficient?

Reply:
'Received properties' sounds a bit confusing. It made me think it is the properties in the incoming message. I think the variable receivedJMSProperties should be renamed to something more descriptive for this specific case. I can only test when there is a new snapshot in nexus and that will probably take time to test.

Reply:
@alisihab, inside the `JMSSender` class and the method `sendMessage` there is a message to be sent: that's logged inside `JMSFacade#send(javax.jms.MessageProducer, javax.jms.Message, boolean)` already. 

The change here at hand, is the waiting of the synchronous response message. That's retrieved at line 238: `javax.jms.Message rawReplyMsg = mc.receive(getReplyTimeout());`
When it is received, all the JMS Properties of the received messaged is now logged as well (in debug mode). Perhaps that's a bit more than the scope of your question, but might be handy too. 
I'll figure out how to get you a new snapshot jar.